--- /home/darinf/minimo/gtk2/pango-1.2.5-build/pangoxft.pc	2003-11-30 02:30:22.000000000 -0800
+++ pangoxft.pc	2003-11-30 03:11:02.000000000 -0800
@@ -6,6 +6,6 @@
 Name: Pango Xft
 Description: Xft font support for Pango
 Version: 1.2.5
-Requires: pango
+Requires: pango xft
 Libs: -L${libdir} -lpangoxft-1.0 
 Cflags: -I${includedir}/pango-1.0 -I/opt/toolchain/arm-linux/local/X11R6/include -I/opt/toolchain/arm-linux/local/X11R6/include/freetype2  
