--- fontconfig.pc	2003-11-29 23:08:23.000000000 -0800
+++ ../../../../local.bak/X11R6/lib/pkgconfig/fontconfig.pc	2003-08-24 21:54:53.000000000 -0700
@@ -1,10 +1,10 @@
-prefix=/usr/X11R6
-exec_prefix=/usr/X11R6/bin
-libdir=/usr/X11R6/lib
-includedir=/usr/X11R6/include
+prefix=/opt/toolchain/arm-linux/local/X11R6
+exec_prefix=${prefix}/bin
+libdir=${prefix}/lib
+includedir=${prefix}/include
 
 Name: Fontconfig
 Description: Font configuration and customization library
 Version: 1.0.1
-Libs: -L${libdir} -lfontconfig
+Libs: -L${libdir} -lfontconfig -L${libdir} -lexpat
 Cflags: -I${includedir}
--- xft.pc	2003-11-29 23:08:25.000000000 -0800
+++ ../../../../local.bak/X11R6/lib/pkgconfig/xft.pc	2003-08-24 23:59:38.000000000 -0700
@@ -1,15 +1,15 @@
-prefix=/usr/X11R6
-exec_prefix=/usr/X11R6/bin
-libdir=/usr/X11R6/lib
-includedir=/usr/X11R6/include
-freetypelibs=-L/usr/X11R6/lib -lfreetype
-freetypecflags=-I/usr/X11R6/include/freetype2
-xrenderlibs=-L/usr/X11R6/lib -lXrender
-xrendercflags=-I/usr/X11R6/include
+prefix=/opt/toolchain/arm-linux/local/X11R6
+exec_prefix=${prefix}/bin
+libdir=${prefix}/lib
+includedir=${prefix}/include
+freetypelibs=-lfreetype
+freetypecflags=-I${includedir}/freetype2
+xrenderlibs=-lXrender
+xrendercflags=
 
 Name: Xft
 Description: X FreeType library
 Version: 2.1.0
 Requires: fontconfig
-Libs: -L${libdir} -lXft ${freetypelibs} ${xrenderlibs}
+Libs: -L${libdir} -lXft -L${libdir} -lXext ${freetypelibs} ${xrenderlibs} -lX11
 Cflags: -I${includedir} ${freetypecflags} ${xrendercflags}
--- xcursor.pc	2003-11-29 23:08:25.000000000 -0800
+++ ../../../../local.bak/X11R6/lib/pkgconfig/xcursor.pc	2003-08-24 20:32:06.000000000 -0700
@@ -1,7 +1,7 @@
-prefix=/usr/X11R6
-exec_prefix=/usr/X11R6/bin
-libdir=/usr/X11R6/lib
-includedir=/usr/X11R6/include
+prefix=/opt/toolchain/arm-linux/local/X11R6
+exec_prefix=${prefix}/bin
+libdir=${prefix}/lib
+includedir=${prefix}/include
 
 Name: Xcursor
 Description: X Cursor library
